### PR TITLE
Skipforback

### LINKED ISSUE #20: Implement Skip Image forward and backward
* Add `SkipForward` and `SkipBack` to Action Enum in `ui.rs`
* In `event_action` function in `ui.rs` map `w` and `PageUp` to SkipForward
* In `event_action` function in `ui.rs` map `b` and `PageDown` to SkipBack
* In `run` in `program.rs` match SkipForward to skip_forward function
* In `run` in `program.rs` match SkipBack to skip_back function
* In new `skip_forward` and `skip_backward` functions, move by 10% of len of images
* If len < 10 move by one image, if 10 < len < 20 move by 2 images etc
* Once skip length is determined, make the call to `increment` or `decrement` with the skip length.

### PR DESCRIPTION
**Pull request by gurgalex**
This implements keyboard shortcuts for skipping through images by a fixed 10% currently.

A couple questions

1. Should the last / first image be shown if the skip increment is too big. 5% left, but want to skip 10%.

2. I assume the stepwise increment will eventually be configurable?

Fixes #20

### CODE REVIEW

File: src/program.rs [hunk 1]
```diff
@@ -108,6 +119,22 @@ impl Program {
         self.render()
     }
 
+    /// Returns new index to advance to
+    pub fn skip_forward(&mut self) -> Result<(), String> {
+        let skip_size = compute_skip_size(&self.images);
+        self.increment(skip_size)?;
+        self.render()?;
```
Comment (Davejkane) [inline src/program.rs]: Increment already includes a call to render. So we can just go with `self.increment(skip_size)` without the `?;` and also get rid of the `Ok(())`

File: src/program.rs [hunk 2]
```diff
@@ -108,6 +119,22 @@ impl Program {
...
+        Ok(())
+    }
+
+    /// Returns new index to skip back to
+    fn skip_backward(&mut self) -> Result<(), String> {
+        let skip_size = compute_skip_size(&self.images);
+        self.decrement(skip_size)?;
+        self.render()?;
```
Comment (Davejkane) [inline src/program.rs]: Same as above.

**Review by Davejkane — CHANGES_REQUESTED** [review]
Looks good. Just a couple of minor changes. As for configuration, we'll take that later. Also, need to fix merge conflicts.

Eventually I'd like to have vim style taking numbers as arguments, so that `10j` will move down 10 images, and `15G` will move to the 15th image. But that's for later. It will require a state machine probably.

**Comment by gurgalex** [comment]
Finally!

File: src/program.rs (hunk 1 above)
Comment (gurgalex) [reply src/program.rs]: Yeah... I keep forgetting that increment and decrement also render the new ui state.

**Comment by gurgalex** [comment]
Ah, right. 
Need to change the logic to go to the last and first image if increment size is larger than length left.

**Comment by gurgalex** [comment]
Hmm, may need direct setting of the index to accomplish that.

Cases to consider

* Decrementing past 0 (should go to index 0)

* Incrementing past index (should cap at last index)

* General increment and decrement

**Review by Davejkane — APPROVED** [review]
Awesome, looks good to me. Can I please request that you don't force push your branch while there is code review because that way I can't see what the latest changes are.

**Pull request merged**